src/Chocolatey.PowerShell/Helpers/PackageParameter.cs

```diff
@@ -0,0 +1,93 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Text.RegularExpressions;
+
+namespace Chocolatey.PowerShell.Helpers
+{
+    public static class PackageParameter
+    {
+        private const string PackageParameterPattern = @"(?:^|\s+)\/(?<ItemKey>[^\:\=\s)]+)(?:(?:\:|=){1}(?:\'|\""){0,1}(?<ItemValue>.*?)(?:\'|\""){0,1}(?:(?=\s+\/)|$))?";
+        private static readonly Regex _packageParameterRegex = new Regex(PackageParameterPattern, RegexOptions.Compiled);
+        
+        public static Hashtable GetParameters(PSCmdlet cmdlet, string parameters)
+        {
+            var paramStrings = new List<string>();
+            var logParams = true;
+            
+            if (!string.IsNullOrEmpty(parameters))
+            {
+                paramStrings.Add(parameters);
+            }
+            else
+            {
+                var packageParameters = EnvironmentHelper.GetVariable(
+                    cmdlet,
+                    "ChocolateyPackageParameters",
+                    EnvironmentVariableTarget.Process);
+                if (!string.IsNullOrEmpty(packageParameters))
+                {
+                    paramStrings.Add(packageParameters);
+                }
+
+                // This should possibly only be implemented in the CLE codebase
+                var sensitivePackageParameters = EnvironmentHelper.GetVariable(
+                    cmdlet,
+                    "ChocolateyPackageParametersSensitive",
+                    EnvironmentVariableTarget.Process);
+                if (!string.IsNullOrEmpty(sensitivePackageParameters))
+                {
+                    logParams = false;
+                    paramStrings.Add(sensitivePackageParameters);
+                }
+            }
+
+            var paramHash = new Hashtable(StringComparer.OrdinalIgnoreCase);
+
+            foreach (var param in paramStrings)
+            {
+                foreach (Match match in _packageParameterRegex.Matches(param))
+                {
+                    var name = match.Groups["ItemKey"].Value.Trim();
+                    var valueGroup = match.Groups["ItemValue"];
+
+                    object value;
+                    if (valueGroup.Success)
+                    {
+                        value = valueGroup.Value.Trim();
+                    }
+                    else
+                    {
+                        value = (object)true;
+                    }
+
+                    if (logParams)
+                    {
+                        cmdlet.WriteDebug($"Adding package param '{name}'='{value}'");
+                    }
```

Comment:
I'm guessing this logic was kind of pre-existing but looking at this I _do_ wonder if we can split things out at some point to log the non-sensitive things and just skip logging sensitive data, maybe.

Reply:
I was wondering the same! I'd happily modify it to do that.